src/screens/MyWorkouts/WorkoutSearchBar.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useEffect, useState, useRef } from 'react';
+import { View, Text, TouchableOpacity } from 'react-native';
+import { ScrollView, TextInput } from 'react-native-gesture-handler';
+import { supabase } from '../../utils/supabaseClient';
+
+/**
+ * @param selectedExercise.exercise //value set to exercise pressed
+ * @param selectedExercise.isOk //if true, then value of exercise is valid, otherwise disregard
+ */
+export type SearchBarWorkoutProps = {
+  selectedExercise: { exercise: string | undefined; isOk: boolean | undefined };
+};
+
+const SearchBarWorkout: React.FC<SearchBarWorkoutProps> = (prop: SearchBarWorkoutProps) => {
+  const [allExercises, setAllExercises] = useState<string[]>([]);
+  const [searchInput, setSearchInput] = useState('');
+  const recentlyUsedExercise = useRef('');
+  const searchArray = useRef<string[]>([]);
+
+  if (searchInput.length > 0) {
+    searchArray.current = [];
+    const tempArr = allExercises.filter((exercise) =>
+      exercise.toLowerCase().includes(searchInput.toLowerCase())
+    );
+    tempArr.map((exercise) => {
+      searchArray.current.push(exercise);
+    });
+  }
+
+  async function fetchExercises() {
```

Comment:
Nit: we could move the exercise fetching logic into a separate hook in the `hooks` folder, but not necessary right now.